### ISSUE TITLE
Decoding _DOMAINS can't handle nested GET-params

### ISSUE THREAD

**Issue by p2media** [dmitryd/typo3-realurl#205]
We are trying to use subdomains for dynamic contents with the following configuration<sup>[1](#footnote1)</sup>:

```php
'_DOMAINS' => array(
	'encode' => array(
		array(
			'GETvar' => 'tx_myext_myplugin[fieldname]',
			'value' => '1',
			'urlPrepend' => 'http://sub1.example.de'
		),
		array(
			'GETvar' => 'tx_myext_myplugin[fieldname]',
			'value' => '2',
			'urlPrepend' => 'http://custom2.example.de'
		)
	),
	'decode' => array(
		'sub1.example.de' => array(
			'GETvars' => array(
				'tx_myext_myplugin[fieldname]' => '1',
			),
		),
		'custom2.example.de' => array(
			'GETvars' => array(
				'tx_myext_myplugin[fieldname]' => '2',
			),
		)
	)
)
```

Currently it's not possible that nested GET-params will be converted properly.

I propose that either `ConfigurationReader::getVarsToSet()` or `UrlDecoder::mergeGetVarsFromDomainsConfiguration()`should do the conversion. We think `UrlDecoder` would be the right place

<a name="footnote1">1</a>: We are generating this configuration of course 😉

**Comment by p2media** [comment]
Just to be clear, this is **not** an issue. It's more a feature request.

**Comment by dmitryd** [comment]
It will take some time. I am going to vacation for the next couple of weeks. So stay patient and wait :) This is definitely useful, so should be implemented...

**Comment by p2media** [comment]
I already wrote a _dirty_ fix. I'll create a pull request soon :)

**Comment by dmitryd** [comment]
Fixed with #206
